api/pkg/imagebuilder/imagebuilder.go

```diff
@@ -615,46 +620,75 @@ func (c *imageBuilder) createKanikoJobSpec(
 		fmt.Sprintf("--dockerfile=%s", baseImageTag.DockerfilePath),
 		fmt.Sprintf("--context=%s", baseImageTag.BuildContextURI),
 		fmt.Sprintf("--build-arg=%s=%s", baseImageEnvKey, baseImageTag.ImageName),
+		fmt.Sprintf("--build-arg=%s=%s", modelArtifactStorageTypeKey, c.artifactService.GetType()),
 		fmt.Sprintf("--build-arg=%s=%s", modelDependenciesUrlEnvKey, modelDependenciesUrl),
 		fmt.Sprintf("--build-arg=%s=%s/model", modelArtifactsUrlEnvKey, version.ArtifactURI),
 		fmt.Sprintf("--destination=%s", imageRef),
 	}
 
+	if c.artifactService.GetType() == "s3" {
+		kanikoArgs = append(
+			kanikoArgs,
+			fmt.Sprintf("--build-arg=%s=%s", "AWS_ACCESS_KEY_ID", os.Getenv("AWS_ACCESS_KEY_ID")),
+			fmt.Sprintf("--build-arg=%s=%s", "AWS_SECRET_ACCESS_KEY", os.Getenv("AWS_SECRET_ACCESS_KEY")),
+			fmt.Sprintf("--build-arg=%s=%s", "AWS_DEFAULT_REGION", os.Getenv("AWS_DEFAULT_REGION")),
+			fmt.Sprintf("--build-arg=%s=%s", "AWS_ENDPOINT_URL", os.Getenv("AWS_ENDPOINT_URL")),
+		)
+	}
```

Comment:
For now I'm just reading this values from the env vars in the environment where the API server in run in. Does anyone have an opinion if we should move it into the config file of the API server? 😅

Reply:
Hmm what do you mean by that?

Reply:
I mean we can move it into config file and add it as kaniko default argument

Reply:
Okay I've removed this entire code chunk, assuming that the user of Merlin will be adding these additional values to the `KanikoAdditionalArgs` [field](https://github.com/caraml-dev/merlin/pull/605/files#diff-7f88f1945fb732d6ac18c0a4b3adfd6bec6f337538050f52d4a52db2e1d65b47R223) of the config file. I've added some additional doc strings [here](https://github.com/caraml-dev/merlin/pull/605/files#diff-7f88f1945fb732d6ac18c0a4b3adfd6bec6f337538050f52d4a52db2e1d65b47R458) to inform them that they'd need to ensure that the kaniko job runner has the necessary credentials to download model artifacts from the blob storage configured.